platforms/jvm/plugins-application/src/main/java/org/gradle/jvm/application/scripts/JavaAppStartScriptGenerationDetails.java

```diff
@@ -37,6 +37,8 @@ public interface JavaAppStartScriptGenerationDetails {
     /**
      * The environment variable to use to control exit value (windows only)
      */
+    @SuppressWarnings("DeprecatedIsStillUsed")
+    @Deprecated
```

Comment:
If you want to deprecate any public (non-internal) methods, you need to use the `DeprecationLogger` to do so. For this method in particular, you'd need to add a deprecation nag to the implementation in `DefaultJavaAppStartScriptGenerationDetails.java`

It would look something like this:
```
    @Override    
    @Deprecated
    @SuppressWarnings("DeprecatedIsStillUsed")
    public String getExitEnvironmentVar() {
        DeprecationLogger.deprecateMethod(JavaAppStartScriptGenerationDetails.class, "getEnvironmentVar()")
            .willBeRemovedInGradle9()
            .withUpgradeGuideSection(8, "remove_exit_env_var")
            .nagUser();
        
        return exitEnvironmentVar;
    }
```

Any other public deprecated method would need further treatment (deprecated methods in `CreateStartScripts`). 

Next, you need to add an entry to `platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc`. Nothing too complicated, but the entry should be something like below. Make sure the properly link to the javadoc for the properties (my below example does not use the correct syntax for that, but there are examples in the file already).
```
[[remove_exit_env_var]]
==== Deprecate exitEnvironmentVar property in CreateStartScripts task
The `getExitEnvironmentVar` and `setExitEnvironmentVar` methods on `CreateStartScripts` and `getExitEnvironmentVar` methods on `JavaAppStartScriptGenerationDetails` have been deprecated and will be removed in Gradle 9.0. 
```

Assuming we get this PR in for Gradle 8.10, this upgrade guide entry should go under the 8.10 section in the upgrade guide. If it does not exist by the time you update this PR, go ahead and create it, following the pattern of the other gradle versions below it in the file (No need to add empty sections for "potential breaking changes", etc). 

Finally, we need integration tests _verifying_ that these deprecation warnings are emitted. I think it would be fine to just add them to `ApplicationPluginIntegrationTest`, which should already have a `CreateStartScripts` task. Something along the lines of:

```
def "exitEnvironmentVar properties are deprecated"() {
    given:
    buildFile << """
        tasks.startScripts {
            // call the getter and setter methods (all 3 deprecated ones)
        } 
    """
 
    expect:
    executer.expectDocumentedDeprecationWarning(<first warning>)
    executer.expectDocumentedDeprecationWarning(<second warning>)
    executer.expectDocumentedDeprecationWarning(<third warning>)
    succeeds(":startScripts")

}
```

Reply:
Couple questions:

- `exitEnvironmentVar` is a property - shouldn't `deprecateProperty` be used instead?
- Default implementation of `JavaAppStartScriptGenerationDetails` is internal - does it make sense to create integration tests for it?